GT-1564 Support managing metatools

frett opened the pull request:
- disable generating drafts and editing pages/tips for metatools
- hide tool settings that don't make sense
- support updating the default variant for a metatool
- don't display metatools on the attachments UI
- support specifying a metatool for other tools

wrandall22 commented on src/app/models/resource.ts:
```diff
@@ -37,6 +37,12 @@ export class Resource {
 
+  static isMetaTool(resource: Resource): boolean {
+    var type = resource.resourceType ? resource.resourceType.name : null;
+    if (type == null) type = resource['resource-type'];
```
Also, should this be combined with the above line?
```
var type = resource.resourceType ? resource.resourceType.name : resource['resource-type'];
```

wrandall22 replied:
Third, I think `type` should be a `const` not a `var`.

frett replied:
the reason I didn't go with
```
var type = resource.resourceType ? resource.resourceType.name : resource['resource-type'];
```

is it would return null if there is a `resourceType`, but the name itself is null. There is a null-coalescing operator that fits a lot better, but that's only available on newer versions of typescript

frett replied:
`==` wasn't intentional, going to update it now

frett replied:
I updated this logic

wrandall22 commented on src/app/components/edit-resource/edit-resource.component.html:
```diff
@@ -174,13 +174,9 @@
+  <div class="row" *ngIf="!isMetaTool()">
     <div class="col">
       <label>Manifest XML</label>
-    </div>
-  </div>
-  <div class="row">
-    <div class="col">
```
Not sure I understand this change.

frett replied:
oh, this was collapsing 2 rows into a single row to match the rest of the layout. It was some cleanup I probably should have done as a separate commit

frett replied:
previously the label and the form field were in 2 separate rows, this combines them into the same row, and just lets them naturally stack within a column in that row.

wrandall22 commented on src/app/components/edit-resource/abstract-edit-resource.component.ts:
```diff
@@ -55,6 +62,10 @@ export abstract class AbstractEditResourceComponent implements OnDestroy {
   }
 
+  compareTools(o1: any, o2: any): boolean {
+    return o1 && o2 && o1.id == o2.id;
```
`===`?

wrandall22 reviewed (changes requested):
Seems like some of the tests are failing.

frett commented:
Updated UIs:

<img width="809" alt="Screen Shot 2022-05-31 at 3 24 54 PM" src="https://user-images.githubusercontent.com/47818/171270522-b60c4ddb-c901-4fad-90ae-f8a40e8f5f61.png">
<img width="801" alt="Screen Shot 2022-05-31 at 3 25 03 PM" src="https://user-images.githubusercontent.com/47818/171270525-f96ff66a-cd97-4b20-bae4-df182924cea9.png">

I moved the OneSky Project Id field down to the row with the hidden and spotlight options so I could put the metatool/variant options on the second row of fields.

wrandall22 commented:
I think all that's left is to fix the tests.
